src/swap.ts

```diff
@@ -0,0 +1,291 @@
+import { purry } from './purry';
+import { isString } from './isString';
+import { isArray } from './isArray';
+import { IterableContainer } from './_types';
+
+type isEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
+
+type isLessThan<A extends number, B extends number> = isEqual<A, B> extends true
+  ? false
+  : 0 extends A
+  ? true
+  : 0 extends B
+  ? false
+  : isLessThan<Subtract<A, 1>, Subtract<B, 1>>;
+
+type BuildTuple<
+  L extends number,
+  T extends ReadonlyArray<unknown> = []
+> = T['length'] extends L ? T : BuildTuple<L, [...T, unknown]>;
+
+type Subtract<A extends number, B extends number> = BuildTuple<A> extends [
+  ...infer U,
+  ...BuildTuple<B>
+]
+  ? U['length']
+  : never;
+
+type IsNonNegativeInteger<T extends number> = number extends T
+  ? false
+  : `${T}` extends `-${string}` | `${string}.${string}`
+  ? false
+  : true;
+
+type Absolute<T extends number> = `${T}` extends `-${infer R}`
+  ? R extends `${infer N extends number}`
+    ? N
+    : never
+  : T;
+
+type StringToChars<T extends string> = string extends T
+  ? Array<string>
+  : T extends `${infer C}${infer R}`
+  ? [C, ...StringToChars<R>]
+  : [];
+
+type Join<
+  Strings extends ReadonlyArray<string>,
+  Accumulator extends string = ''
+> = Strings extends readonly [infer Head, ...infer Rest]
+  ? Rest extends ReadonlyArray<string>
+    ? Join<
+        Rest,
+        Head extends string
+          ? Accumulator extends ''
+            ? Head
+            : `${Accumulator}${Head}`
+          : never
+      >
+    : Accumulator
+  : Accumulator;
```

Comment:
I just remembered we already have our version of the join type, checkout our join function!

Export the Joined type there and use it here...

https://github.com/remeda/remeda/blob/master/src/join.ts#L6-L20